code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -1,17 +1,45 @@
 /obj/item/reagent_containers/food/drinks/cans
-	var/canopened = 0
+	var/canopened = FALSE
 	var/is_glass = 0
 	var/is_plastic = 0
+	var/shaken = 0
+	var/bursting = FALSE
+	var/burstchance = 0
 
 /obj/item/reagent_containers/food/drinks/cans/New()
 	..()
 	flags &= ~OPENCONTAINER
 
-/obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user)
-	if(canopened == 0)
+/obj/item/reagent_containers/food/drinks/cans/attack_self(mob/user, burstopen = FALSE)
```

Comment:
Can you make this one `return ..()`. I did notice that `drinks/attack_self` does not `return ..()`... which prevents it from sending the attackself signal but hey... better ensure this one works when that one gets fixed